articles/performance-optimization-techniques-for-node-js/index.md

```diff
@@ -0,0 +1,390 @@
+
+### Query optimization
+Just imagine having more than one million rows of data to query from your database table, the process generated by this database to get the endpoint of this data will greatly affect your application performance, and displaying this data can make it even slower. Using bad queries will make your application almost useless (and will leave your users dissatisfied). The solution to this is to perform a query optimization.
+
+#### Basic tips to improve your database performance/[optimization overview](https://dev.mysql.com/doc/refman/8.0/en/optimize-overview.html)
+
+- Database and table indexing/structure. Proper [indexing](https://dev.mysql.com/doc/refman/8.0/en/optimization-indexes.html) will speed up data retrieval between your databases and tables to select and sort rows faster. Proper indexing will help you avoid columns from overlapping.
+
+- Avoid [SELECT](https://dev.mysql.com/doc/refman/8.0/en/select-optimization.html)
+
+This would be the most obvious thing you would use to select your columns, right? Yet, you should not overlook how cumbersome this query can be, to list down columns you have in a table. Go for a more distinct approach using the SELECT statement to query only the data you need and avoid extra fetching loads to your database.
+
+- Use the [clause WHERE](https://dev.mysql.com/doc/refman/8.0/en/where-optimization.html)
+
+The goal of the query is to pull the required records from your database. WHERE clause helps to filter the records and limits the number of records to be pulled based on conditions. WHERE replaces the HAVING clause that selects records before filtering the dataset. In SQL operations, WHERE statements are faster as it reduces that data being processed by the database engine.
+
+- [Use LIMIT](https://dev.mysql.com/doc/refman/8.0/en/limit-optimization.html) to sample the query results
+
+LIMIT will return only the specified number of records. While using LIMIT, ensure that the results are desirable and meaningful when a limit is imposed on the dataset.
+
+- To avoid columns use [text joins](https://dev.mysql.com/doc/internals/en/optimizer-joins.html)
+
+Joins with strings/tests are considered much slower when compared to numeric columns. The solution to this is to always make sure foreign key strings have an equivalent numeric representation where join can be performed.
+
+Here is an example:
+
+```sql
+SELECT from cars
+WHERE color = 'red'
+```
+```sql
+SELECT from cars
+WHERE color_id = 2
+```
+
+The second query will execute faster than the first query due to the numeric representation.
+
+- Running queries during off-peak hours
+
+In the production database, analytical and database management queries should be executed when the concurrent users are at their lowest peak. Typically at night around 3 to 5 am.
+
+#### Other Tips
+- Avoid SELECT DISTRICT
```

Comment:
why not? add a short sentence (like the other bullet points) why should I avoid SELECT DISTRICT